### PR TITLE
layers: Fix some warnings

### PR DESCRIPTION
**Pull request by krOoze**

### CODE REVIEW

**Comment by krOoze** [comment]
Rebased and removed a change offending the CI for now.

File: layers/vk_layer_utils.cpp [hunk 1]
```diff
@@ -149,62 +150,6 @@ VK_LAYER_EXPORT void layer_debug_messenger_actions(debug_report_data *report_dat
     }
 }
 
-// NOTE: This function has been deprecated, and the above function (layer_debug_messenger_actions) should be
```
Comment (mark-lunarg) [inline layers/vk_layer_utils.cpp]: This function and it's support functions are used in other repos (VulkanTools at least). Can we leave it in for now?

Comment (krOoze) [reply layers/vk_layer_utils.cpp]: Alright, commit dropped.

Comment (krOoze) [reply layers/vk_layer_utils.cpp]: PS: though I think one of the CI compiles Vulkan-Tools, and there were no problems. Except the one with #1375.

**Comment by mark-lunarg** [comment]
@krOoze, this is all fine as well, but after going through CI here it looks like the renaming in `layers: Restyle vk_layer_config` will also hose the downstream repos.  You could pull that commit into the other PR.  If it's OK with you, I can just merge the other three for now.  Lemme know what you'd prefer.

**Comment by mark-lunarg** [comment]
BTW, travis and appveyor pass because the known_good in the downstream repos has not been updated to contain the changes in this PR -- this usually happens at header-update or sdk-build time.

**Comment by krOoze** [comment]
@mark-lunarg Only the header changes should cause problems, I think. I will just extract that stuff out momentarily.

**Comment by krOoze** [comment]
@mark-lunarg done

**Pull request merged**